### PR TITLE
Simplify code, follow linter suggestions

### PR DESCRIPTION
**Pull request by matiasinsaurralde**

### CODE REVIEW

File: core/crypto/signatures/p256/ecdsa_p256_test.go [hunk 1]
```diff
@@ -230,7 +230,7 @@ func flipBit(a []byte, pos uint) []byte {
 
 func TestGeneratePEMs(t *testing.T) {
 	signatures.Rand = DevZero{}
-	skPEM, pkPEM, err := GeneratePEMs()
```
Comment (cesarghali) [inline core/crypto/signatures/p256/ecdsa_p256_test.go]: Please keep this error, check it, and stop the test with `t.Fatalf` if it's not nil.

Comment (matiasinsaurralde) [inline core/crypto/signatures/p256/ecdsa_p256_test.go]: @cesarghali Did a rebase and it was already there!

**Comment by matiasinsaurralde** [comment]
@cesarghali I'm using [gosimple](https://github.com/dominikh/go-simple), [errcheck](https://github.com/kisielk/errcheck), vetshadow (`go tool vet --shadow *.go`)

**Comment by gdbelvin** [comment]
@matiasinsaurralde Thanks for this PR. Would you mind splitting this into three PRs, one for the changes requested by each tool?  

Thanks!

**Comment by matiasinsaurralde** [comment]
@gbdelvin Sure, will do it today

**Comment by gdbelvin** [comment]
Looking forward to your PRs :-)

**Comment by gdbelvin** [comment]
Closing due to staleness.  Please reopen if further changes are planned.

**Pull request closed without merging**